[SYCL][DOC] Document test-related processes (#2760)
intel/llvm: sycl/test/on-device/README.md

# DPC++ end-to-end tests for features under development

> It is recommended to have in-tree LIT tests independent from HW (e.g. GPU,

Reviewer: ```suggestion

We require to have in-tree LIT tests independent from HW (e.g. GPU,
```
